tests/specialization_constants/specialization_constants_exceptions_catching_fp64.cpp

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for specialization constants expected exceptions catching for
+//  double
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+
+#include "specialization_constants_exceptions_catching_common.h"
+
+#define TEST_NAME specialization_constants_exceptions_catching_fp64
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+
+/** test specialization constants expected exceptions catching for double
+ */
+class TEST_NAME : public util::test_base {
+ public:
+  /** return information about this test
+   */
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    try {
+      auto queue = util::get_cts_object::queue();
+      if (!queue.get_device().has(sycl::aspect::fp64)) {
+        log.note(
+            "Device does not support double precision floating point "
+            "operations");
+        return;
+      }
+#ifndef SYCL_CTS_FULL_CONFORMANCE
+      check_spec_constant_except_catch_for_type<double> fp64_test{};
+      fp64_test(log, "double");
+#else
+      for_type_vectors_marray<check_spec_constant_except_catch_for_type,
+                              double>(log, "double");
+#endif
+
+    } catch (const sycl::exception &e) {
+      log_exception(log, e);
+      std::string errorMsg =
+          "a SYCL exception was caught: " + std::string(e.what());
+      FAIL(log, errorMsg);
+    } catch (const std::exception &e) {
+      std::string errorMsg =
+          "an exception was caught: " + std::string(e.what());
+      FAIL(log, errorMsg);
+    }
+  }
+};
+
+// construction of this proxy will register the above test
+util::test_proxy<TEST_NAME> proxy;
+
+}  // namespace TEST_NAMESPACE
```

Comment:
A lot of code duplication across the 3 different tests.
Do you think it is possible to factorize most of it into `specialization_constants_exceptions_catching_common.h`?

Reply:
Ronan, are you talking about try-catch blocks?
this blocks will be moved to test base, we talking about this in PR#140